core/domain/feedback_domain.py

```diff
@@ -162,11 +128,11 @@ class FeedbackMessage(object):
     """
 
     def __init__(
-            self, full_message_id, full_thread_id, message_id, author_id,
+            self, full_message_id, thread_id, message_id, author_id,
```

Comment:
Does full_message_id also need to be refactored (when thinking about the general case)?

(Also, is it actually used?)

Reply:
message_id is actually interesting. I had a discussion with Anmol about this. So basically message_id (partial) is an incremental int. the full_message_id is thread_id.int. This actually made sense to me. Messages being part of a thread and can be identified by the int as such when you know the thread_id. The key for the model is the full id however (situation similar to feedback threads). 

For the general case -- Because thread_id will change, the corresponding models that store thread_id as a key/part of the key need to be migrated.

Reply:
Yep, I agree thread_id.int sounds reasonable.

But do you actually need to store the full_message_id param separately? That makes it seem like it's independent of thread_id and message_id, whereas in reality it's not. Couldn't it be constructed on the fly when needed (via a @property, or a method in the domain class?)

Reply:
I think yes it can be constructed on the fly. Shall I follow up with a PR for that (already this one seems too hard to test and verify).

Reply:
Yes, separate PR is fine for this; file an issue though since it's a codebase deficiency. Thanks!

Reply:
Linked it to the same issue I filed with future steps.